### DISCUSSION TITLE
Bring minimum commissions online asap instead of waiting for cosmwasm to be ready

### DISCUSSION THREAD

**Discussion by Terra854** [crypto-org-chain/chain-main#861] [Ideas]
With the upgrade of the cosmos-sdk to 0.46, minimum commissions has finally been implemented inside the codebase. As such I would like to suggest that we carry out an upgrade to the network asap so that we can bring the feature online instead of having to wait for the next phase of the roadmap (cosmwasm support) to be ready to bundle minimum commissions with it.

**Comment by zkaizhi** [comment]
Hi @Terra854,

I think this is indeed something we are actively looking into supporting, and agree that this is vital for long-term sustainability of the infrastructure in our ecosystem.

A summary of what has been discussed until now:

- A Min commission of 5% is in line with what other cosmos ecosystem chains have implemented, e.g. evmos, osmosis, juno etc.
- A min commission of 5% will redistribute the reward emissions with approximately 3%. This means roughly:
for delegators from 94% -> 91%
for validators from 5.8% -> 8.6%
- We are looking at cosmos hub [proposal 76](https://wallet.keplr.app/chains/cosmos-hub/proposals/76), which could be an indicator for our proposal. This signalling vote has however been rejected, which may or may not have been due to the format of the proposal. More details on the discussion can be found [here](https://forum.cosmos.network/t/proposal-76-vote-on-chain-require-a-minimum-5-commission-for-all-validator-nodes/7103).

Nevertheless, it is still the idea to do a signalling vote to capture user feedback before we go into the final proposal format.

Feel free to provide any feedback or suggestions on this proposal here in this discussion.

**Reply by zanglang** [reply]
> for delegators from 94% -> 91% ... for validators from 5.8% -> 8.6%

@zkaizhi could you share the math on how this was derived as well?

**Reply by zkaizhi** [reply]
Basically
1. For the current top 100 validators, we calculate the current theoretical APY rewards, 
which is calculated by (Annual Provisions / total bonded amount) x (estimated block per year / block per year from genesis)
2. Taking into account their current commission fee, now calculate the % of this rewards going to validator and % going to delegator
3. Now re-calculate the same, but adjust the commission fee with a minimum set to 5%, so this basically only impacts validators with a lower than 5%, commission fee.

**Reply by Terra854** [reply]
So you prefer we ask others about implementing min comms before we actually do it?

**Reply by zkaizhi** [reply]
> So you prefer we ask others about implementing min comms before we actually do it?

Oh what I mean is that, we already see that other chains in the community have been doing something similar, so it's definitely what we are also looking at doing. And the discussion is just when and what will be the proper format raising this proposal

**Reply by Terra854** [reply]
Oh ok, i think we can start drafting the proposal now since the next version of chain-maind which have the min commission baked in is now in beta
